packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.kt

```diff
@@ -169,18 +169,21 @@ open class FileDownloader(context: Context) {
       return
     }
 
-    parseManifest(manifestBodyAndHeaders.first, manifestBodyAndHeaders.second, extensions, configuration, callback)
+    // part headers take precedence over main response headers
+    // for example, content-type should be that of the part (application/json), not multipart/*
+    val combinedHeaders = Headers.Builder().addAll(response.headers()).addAll(manifestPartBodyAndHeaders.second).build()
+    parseManifest(manifestPartBodyAndHeaders.first, combinedHeaders, extensions, configuration, callback)
   }
 
-  private fun parseManifest(manifestBody: String, manifestHeaders: Headers, extensions: JSONObject?, configuration: UpdatesConfiguration, callback: ManifestDownloadCallback) {
+  private fun parseManifest(manifestBody: String, allManifestResponseHeaders: Headers, extensions: JSONObject?, configuration: UpdatesConfiguration, callback: ManifestDownloadCallback) {
```

Comment:
Maybe "combinedManifestResponseHeaders", if you find that clearer. (This thought occurred to me before I saw you named the stack variable `combinedHeaders`, which makes me think we independently arrived at a similar conclusion.)

```suggestion
  private fun parseManifest(manifestBody: String, combinedManifestResponseHeaders: Headers, extensions: JSONObject?, configuration: UpdatesConfiguration, callback: ManifestDownloadCallback) {
```